enterprise/cypher/cypher-compiled-runtime-3.2/src/test/scala/org/neo4j/cypher/internal/compiled_runtime/v3_2/SaveGeneratedSource.scala

```diff
@@ -0,0 +1,86 @@
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compiled_runtime.v3_2
+
+import java.nio.file.FileVisitResult.CONTINUE
+import java.nio.file._
+import java.nio.file.attribute.BasicFileAttributes
+
+import org.neo4j.cypher.internal.compiled_runtime.v3_2.SaveGeneratedSource.GENERATED_SOURCE_LOCATION
+import org.neo4j.cypher.internal.frontend.v3_2.test_helpers.CypherTestSupport
+
+/**
+  * This trait allows debugging generated queries, by generating queries through java source, then making sure that
+  * the generated source is written to a place where Intellij can find it.
+  *
+  * How to use:
+  * 1. Add this trait to the test class containing your test.
+  * 2. Prefix your query with `CYPHER debug=generate_java_source`
+  * 3. When running the test make sure that the Working Directory is set to
+  *    the directory of the maven module containing your test.
+  * 4. Mark `[your-maven-module]/target/generated-test-sources/cypher` as "Generated Sources Root".
+  * 5. Make sure you have a breakpoint set to somewhere before execution enters the generated code,
+  *    but after the code has been generated.
+  * 6. When you run you test, as the first breakpoint triggers, find the directory
+  *    `[your-maven-module]/target/generated-test-sources/cypher`, right click and select "Synchronize 'cypher'"
+  *    If you have not done so before, this is a good time to "Mark Directory as" "Generated Sources Root".
+  * 7. Now you should see the source file for the generated query, and be able to set breakpoints in that code,
+  *    as well as stepping through it.
+  * 8. Note that every time you re-run your test, you will have to repeat steps 5 to 7, since new code will be
+  *    generated each time.
+  */
+trait SaveGeneratedSource extends CypherTestSupport {
+  private var generatedSources: Option[Path] = None
+
+  override protected def initTest(): Unit = {
+    super.initTest()
+    val cwd = Paths.get(".").normalize.toRealPath()
+    // If CWD is set up correctly, we assign the generated source location
+    if(Files.isRegularFile(cwd.resolve("src/test/scala").resolve(getClass.getName.replace('.', '/') + ".scala"))
+      && Files.isDirectory(cwd.resolve("target"))) {
+      setLocation(cwd.resolve("target").resolve("generated-test-sources").resolve("cypher"))
+    }
+  }
+
+
+  private def setLocation(location: Path) = {
+    System.err.println(s"Will save generated sources to $location")
```

Comment:
No, it's actually there on purpose, and for two reasons:

1. To remind you of where the source files are saved to - to help with steps 4-6, and
2. To allow you to verify that you've set up your Working Directory property, if you see that printout - you know that you did it right.